### ISSUE TITLE
Extension Manager is not consistent it seems

### ISSUE THREAD

**Issue by zounp** [splitbrain/dokuwiki#2965]
Release 2018-04-22b "Greebo"
I did in Extension Manager in the tab "Search and Install" a search for "color" several times. The results were not consistent. One time I found the "Color Syntax Plugin" (https://www.dokuwiki.org/plugin:color) and not the "Color Plugin" (no url from this one saved so far). On an other occasion I found the "Color Plugin" and not the "Color Syntax Plugin".
I am not able to reproduce it. The latest searches produce the "Color Syntax Plugin" and not the "Color Plugin".
When I search for "color syntax" the search yields no results which is strange since searching for "color" yields "Color Syntax Plugin"!
The name of the "Color Syntax Plugin"  on https://www.dokuwiki.org/plugin:color is "Color Plugin" which seems strange to me since I would expect "Color Syntax Plugin".
See the screenshots.
![Search for color - Color Plugin found](https://user-images.githubusercontent.com/19635343/72687740-ca62e280-3af8-11ea-907a-403ff8a10bb4.png)
![Search for color - Color Syntax Plugin found](https://user-images.githubusercontent.com/19635343/72687741-cafb7900-3af8-11ea-8289-cca8a730d7e7.png)
![Search for color syntax - Color Syntax Plugin not found](https://user-images.githubusercontent.com/19635343/72687742-cb940f80-3af8-11ea-9a58-be69799b2466.png)

This probably needs more investigation.

**Comment by phy25** [comment]
Duplicate of #2024.

- https://www.dokuwiki.org/plugin:color `color Plugin`
- https://github.com/hanche/dokuwiki_color_plugin/blob/master/plugin.info.txt#L5 `color syntax plugin`

If you have any further suggestions please comment and I can reopen this.

**Comment by zounp** [comment]
Maybe it is partially the same. Getting different search results is not mentioned in #2024. The search for "Color Syntax" is maybe indirect mentioned in #2024. 
I agree with #2024, it should be consistent. As it is it can be confusing.

**Comment by phy25** [comment]
Generally the behaviour you see is because once you have the plugin installed, the installation page will show the local name of the plugin, not the one from the server.

**Comment by phy25** [comment]
I think this may be a deliberate decision because you may have plugins with the same name installed but from a different source, and your own installation always has the priority. As you may notice, the longer name you are seeing in installation page is actually showing an installed status.

I reopened the issue but I highly doubt if we will fix this.

**Comment by zounp** [comment]
All the plugins installed on my Dokuwiki instance are installed via the "Extension Manager".

Maybe an other issue:
When I click on "Install" in the "Color Plugin" case, which I am getting now as search result, Dokuwiki returns a blank screen. I have to restore from backup, putting public_html back, to get Dokuwiki working again.
After that when I click on "Extension Manager" tab "Installed Plugins" I get no results although plugins are installed.

**Comment by phy25** [comment]
@zounp The issue you mentioned is another issue, please do not mix them together. You may want to use https://forum.dokuwiki.org/ instead for user support.

I am not actually quite familiar with extension manager related issue, maybe there are some cache and metadata involved.

**Comment by zounp** [comment]
> @zounp The issue you mentioned is another issue, please do not mix them together. You may want to use https://forum.dokuwiki.org/ instead for user support.
Yes, you are right about it being another issue. I do not know if it is a support issue when I install a plugin form "Extension Manager" and the Dokuwiki instance gets corrupted. I needs some more investigation though. Later more.
Continued in issue #2966.

**Comment by Klap-in** [comment]
The data is from  different sources. It’s shown by the https://www.dokuwiki.org/plugin:extension plugin, part of the info is delivered by https://github.com/splitbrain/dokuwiki-plugin-pluginrepo plugin hosted at DokuWiki.org, and part from plugin.info.txt written by the plugin author.
Maybe making a list via pluginrepo could already show the inconsistencies. And these insights could help others to improve the data.

**Comment by phy25** [comment]
From the extension manager perspective, I think it makes sense to force `tab=search` to always show `$remoteInfo` by adding a `$forceSource` variable and adding checks on those output methods. This should be sufficient to reduce confusion (and close this issue).
